### PR TITLE
list view multi edit

### PR DESCRIPTION
**Pull request by aab-odoo**

### CODE REVIEW

File: addons/web/static/src/views/list/list_confirmation_dialog.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
                                 <tr>
                                     <td>Update to:</td>
                                     <td style="pointer-events: none;">
-                                        <Field name="field.name" record="props.record" type="field.widget" readonly="true" fieldInfo="props.fieldNodes[field.name]"/>
+                                        <Field name="field.name" record="props.record" type="field.fieldNode.widget" readonly="true" fieldInfo="field.fieldNode"/>
```
Comment (aab-odoo) [inline addons/web/static/src/views/list/list_confirmation_dialog.xml:26]: fieldNode, fieldNodes, fieldInfo... all these refer to the same thing, we should clarify the terminology

File: addons/web/static/src/views/list/list_controller.js [hunk 2]
```diff
@@ -516,22 +513,23 @@ export class ListController extends Component {
...
+                        this.model.root.leaveEditMode({ discard: true });
                         resolve(false);
                     },
                     isDomainSelected,
-                    fields: Object.keys(editedRecord.getChanges()).map((fieldName) => {
-                        const activeField = editedRecord.activeFields[fieldName];
+                    fields: Object.keys(changes).map((fieldName) => {
+                        const fieldNode = Object.values(this.archInfo.fieldNodes).find(
+                            (fieldNode) => fieldNode.name === fieldName
+                        );
```
Comment (FrancoisGe) [inline addons/web/static/src/views/list/list_controller.js:523]: multiple occurrences of the same field ?

File: addons/web/static/src/views/list/list_confirmation_dialog.xml (hunk 1 above)
Comment (pro-odoo) [reply addons/web/static/src/views/list/list_confirmation_dialog.xml:26]: `fieldDefinition` ?

**Review by pro-odoo — APPROVED** [review]

File: addons/web/static/src/views/relational_model/dynamic_list.js [hunk 3]
```diff
@@ -203,4 +207,51 @@ export class DynamicList extends DataPoint {
             return this._load(0, this.limit, this.orderBy);
         }
     }
+
+    async _multiSave(record) {
+        const changes = record._getChanges();
+        if (!Object.keys(changes).length) {
+            return;
+        }
+        const { _onWillSaveMultiRecords, _onSavedMultiRecords } = this.model;
```
Comment (ged-odoo) [inline addons/web/static/src/views/relational_model/dynamic_list.js]: why this? wouldnt it be simpler to just to `this.model._onWillSaveMultiRecords(...)` when needed?

File: addons/web/static/src/views/relational_model/relational_model.js [hunk 4]
```diff
@@ -108,9 +108,14 @@ export class RelationalModel extends Model {
         this.defaultGroupBy = params.defaultGroupBy;
         this.maxGroupByDepth = params.maxGroupByDepth;
         this.groupByInfo = params.groupByInfo || {};
-
-        this._onWillSaveRecord = params.onWillSaveRecord || (() => {});
-        this._onRecordSaved = params.onRecordSaved || (() => {});
+        this.multiEdit = params.multiEdit;
+
+        const hooks = params.hooks || {};
```
Comment (ged-odoo) [inline addons/web/static/src/views/relational_model/relational_model.js]: maybe we could keep the hooks object?

File: addons/web/static/src/views/list/list_controller.js [hunk 5]
```diff
@@ -237,12 +239,7 @@ export class ListController extends Component {
     }
 
     onClickDiscard() {
-        const editedRecord = this.model.root.editedRecord;
-        if (editedRecord.isNew) {
-            this.model.root.removeRecord(editedRecord);
-        } else {
-            editedRecord.discard();
-        }
+        this.model.root.leaveEditMode({ discard: true });
```
Comment (ged-odoo) [inline addons/web/static/src/views/list/list_controller.js:242]: any reason why `leaveEditMode` is not on model itself? Also, i am not sure I like the method name. Shouldn't it be `setReadonly` or something that explains the goal of the method?

File: addons/web/static/src/views/relational_model/dynamic_list.js [hunk 6]
```diff
@@ -203,4 +207,51 @@ export class DynamicList extends DataPoint {
...
+            const context = this.context;
+            try {
+                await this.model.orm.write(this.resModel, resIds, changes, { context });
+            } catch (e) {
+                record._discard();
+                this.model._updateConfig(record.config, { mode: "readonly" }, { noReload: true });
+                throw e;
+            }
+            const records = await this.model._loadRecords({ ...this.config, resIds });
+            validSelection.forEach((record) =>
```
Comment (ged-odoo) [inline addons/web/static/src/views/relational_model/dynamic_list.js]: for of

**Review by ged-odoo — APPROVED** [review]

**Pull request merged**